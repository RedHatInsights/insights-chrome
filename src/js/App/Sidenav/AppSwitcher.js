import React, { useState } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Dropdown } from '@patternfly/react-core/dist/js/components/Dropdown/Dropdown';
import { DropdownItem } from '@patternfly/react-core/dist/js/components/Dropdown/DropdownItem';
import { DropdownToggle } from '@patternfly/react-core/dist/js/components/Dropdown/DropdownToggle';
import { CaretDownIcon } from '@patternfly/react-icons/dist/js/icons/caret-down-icon';

import './AppSwitcher.scss';

const AppSwitcher = ({ currentApp }) => {

    const [isOpen, setIsOpen] = useState(false);

    const dropdownItems = (activeApp) => {

        const appList = [
            { title: 'Red Hat Insights', id: 'insights' },
            { title: 'Red Hat OpenShift Cluster Manager', id: 'openshift' },
            { title: 'Red Hat Ansible Automation Platform', id: 'ansible' },
            { title: 'Cost Management', id: 'cost-management' },
            { title: 'Migration Services', id: 'migrations' },
            { title: 'Subscription Watch', id: 'subscriptions' }
        ];

        const renderNavItems = appList.map(app =>
            <DropdownItem
                component='a'
                href={`${document.baseURI}${app.id}`}
                className={classNames({ 'ins-c-app-switcher__current': app.title === activeApp })}
                key={app.id}>
                {app.title}
            </DropdownItem>
        );

        return renderNavItems;

    };

    return (
        <section className='ins-c-app-switcher'>
            <Dropdown
                isPlain
                className='ins-c-app-switcher__dropdown'
                onSelect={() => setIsOpen(!isOpen)}
                toggle={
                    <DropdownToggle id="toggle-id" onToggle={() => setIsOpen(!isOpen)} toggleIndicator={CaretDownIcon}>
                        { currentApp }
                    </DropdownToggle>
                }
                isOpen={isOpen}
                dropdownItems={dropdownItems(currentApp)}
            />
        </section>
    );
};

export default AppSwitcher;

AppSwitcher.propTypes = {
    currentApp: PropTypes.string
};
