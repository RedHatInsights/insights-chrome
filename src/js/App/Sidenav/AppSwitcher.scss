.ins-c-app-switcher {
    width: 100%;
    padding: calc(8px + var(--pf-global--spacer--sm)) 12px;
    margin-top: 7px;
    border-bottom: 3px solid #3c3f42;

    .ins-c-app-switcher__dropdown {
        width: 100%;
        padding: 8px 12px;
        &.pf-m-expanded, &:hover {
            background: #393c3d;
        }

        .pf-c-dropdown__menu {
            min-width: unset;
            left: 0;
        }

        .pf-c-dropdown__toggle {
            color: white;
            padding: 0;
            width: 100%;
            justify-content: space-between;
            .pf-c-dropdown__toggle-text {
                font-weight: var(--pf-global--FontWeight--semi-bold);
            }
        }
        
        .pf-c-dropdown__menu-item {
            word-wrap: break-word;
            white-space: normal;
            &.ins-c-app-switcher__current {
                font-weight: var(--pf-global--FontWeight--semi-bold);
                background: var(--pf-c-dropdown__menu-item--hover--BackgroundColor);
            }
        }
    }
}