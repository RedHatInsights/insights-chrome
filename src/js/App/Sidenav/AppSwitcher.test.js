import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import AppSwitcher from './AppSwitcher';

describe('AppSwitcher', () => {

    it('should render correctly', () =>{
        const wrapper = shallow(<AppSwitcher currentApp='Red Hat Insights'/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
