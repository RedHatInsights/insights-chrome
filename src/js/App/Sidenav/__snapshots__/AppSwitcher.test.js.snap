// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSwitcher should render correctly 1`] = `
<section
  className="ins-c-app-switcher"
>
  <Component
    className="ins-c-app-switcher__dropdown"
    dropdownItems={
      Array [
        <Unknown
          className="ins-c-app-switcher__current"
          component="a"
          href="http://localhost/insights"
        >
          Red Hat Insights
        </Unknown>,
        <Unknown
          className=""
          component="a"
          href="http://localhost/openshift"
        >
          Red Hat OpenShift Cluster Manager
        </Unknown>,
        <Unknown
          className=""
          component="a"
          href="http://localhost/ansible"
        >
          Red Hat Ansible Automation Platform
        </Unknown>,
        <Unknown
          className=""
          component="a"
          href="http://localhost/cost-management"
        >
          Cost Management
        </Unknown>,
        <Unknown
          className=""
          component="a"
          href="http://localhost/migrations"
        >
          Migration Services
        </Unknown>,
        <Unknown
          className=""
          component="a"
          href="http://localhost/subscriptions"
        >
          Subscription Watch
        </Unknown>,
      ]
    }
    isOpen={false}
    isPlain={true}
    onSelect={[Function]}
    toggle={
      <Unknown
        id="toggle-id"
        onToggle={[Function]}
      >
        Red Hat Insights
      </Unknown>
    }
  />
</section>
`;
